src/components/grid/Grid.jsx

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useStateValue } from '../../app/state/useStateValue';
+import { constants } from '../../app/config';
+import Row from '../row/Row';
+import './Grid.css';
+
+export default function Grid () {
+  const [state, dispatch] = useStateValue();
+  const [columns, setColumns] = useState(0);
+  const [rows, setRows] = useState(0);
+  const running = state.gameStatus.running;
+
+  const renderGrid = () => {
+    const removeHeight = 85;
+    const removeWidth = 10;
+    const columns =
+      Math.floor((window.innerWidth - removeWidth) / constants.CELL_SIZE[0]) -
+      1;
+    const rows =
+      Math.floor((window.innerHeight - removeHeight) / constants.CELL_SIZE[1]) -
+      1;
+    setColumns(columns);
+    setRows(rows);
+    dispatch({
+      type: 'SET_GRID',
+      payload: {
+        columns,
+        rows
+      }
+    });
+  };
+
+  const resizeHandler = useCallback(() => {
+    // block change while running
+    if (running) {
+      return;
+    }
+
+    renderGrid();
+  }, [running]);
+
+  useEffect(() => {
+    renderGrid();
+    window.addEventListener('resize', resizeHandler);
+    return () => {
+      window.removeEventListener('resize', resizeHandler);
+    };
+  }, [resizeHandler]);
+
+  return (
+    <div className='grid'>
+      <div className='columnLabel'>
+        {columns > 0 &&
+          [...Array(columns + 1).keys()].map((e, i) => (
```

Comment:
`Array.from` would be more suitable here (better readability, performance & directly iterable without the need for a spread)
(same in `Row.jsx`)

Reply:
another good recommendation
fixed